kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/TypedClusterScopeCustomResourceApiTest.java

```diff
@@ -139,9 +140,28 @@ void testStatusUpdate() throws InterruptedException {
 
       starClient.inNamespace("test").updateStatus(updatedStar);
       RecordedRequest recordedRequest = server.getLastRequest();
+      assertEquals(1, server.getRequestCount());
       assertEquals("PUT", recordedRequest.getMethod());
-      assertEquals("{\"apiVersion\":\"example.crd.com/v1alpha1\",\"kind\":\"Star\",\"metadata\":{\"name\":\"sun\"},\"spec\":{\"type\":\"G\",\"location\":\"Galaxy\"},\"status\":{\"location\":\"M\"}}", recordedRequest.getBody().readUtf8());
-      System.out.println(recordedRequest.getBody().readUtf8());
+      assertEquals("{\"apiVersion\":\"example.crd.com/v1alpha1\",\"kind\":\"Star\",\"metadata\":{\"name\":\"sun\",\"resourceVersion\":\"1\"},\"spec\":{\"type\":\"G\",\"location\":\"Galaxy\"},\"status\":{\"location\":\"M\"}}", recordedRequest.getBody().readUtf8());
+    }
+
+    @Test
+    void testStatusReplace() throws InterruptedException {
+      Star updatedStar = getStar();
+      StarStatus starStatus = new StarStatus();
+      starStatus.setLocation("M");
+      updatedStar.setStatus(starStatus);
+
+      server.expect().put().withPath("/apis/example.crd.com/v1alpha1/stars/sun/status").andReturn(200, updatedStar).once();
+      server.expect().get().withPath("/apis/example.crd.com/v1alpha1/stars/sun").andReturn(200, "{\"apiVersion\":\"example.crd.com/v1alpha1\",\"kind\":\"Star\",\"metadata\":{\"name\":\"sun\",\"resourceVersion\":\"1\"},\"spec\":{\"type\":\"G\",\"location\":\"Galaxy\"},\"status\":{\"location\":\"M\"}}").once();
```

Comment:
Why do you use an `Star` in the first expectation and a `String` in the later?

Reply:
I've updated to just use strings and be more readable.